### DISCUSSION TITLE
Filenames without extension in package.json

### DISCUSSION THREAD

**Discussion by alemuller** [tree-sitter/tree-sitter#1083] [Q&A] [answered]
In package.json, filenames are usually specified as:

```json
"tree-sitter": [
{
  "scope": "source.mk",
  "file-types": [
    "mk",
    "MAKEFILE"
  ]
}
```

How can I specify `scope` for multiples filenames without extension, such `MAKEFILE`, `GNUMAKEFILE`, `Makefile`, etc. And also support filenames with extension `.mk`?

**Comment by maxbrunsfeld** [comment, marked as answer]
I think you can just put this:

```json
{
  "tree-sitter": [
    {
      "scope": "source.mk",
      "file-types": [
        "mk",
        "Makefile",
        "MAKEFILE",
        "GNUMAKEFILE",
      ]
    }
  ]
}
```

A `file-types` entry can be either be a complete filename or a filename extension.
